overlord/ifacestate/ifacestate.go

```diff
@@ -36,11 +36,6 @@ import (
 	"github.com/snapcore/snapd/snap"
 )
 
-var noConflictOnConnectTasks = func(task *state.Task) bool {
```

Comment:
@stolowski should double check this one, maybe a in-flight PR uses it?

Reply:
It's fine, safe to remove, I forgot about it last time I touched conflict checking in ifacestate. Thanks!

Reply:
it was used by less streamlined/conservative conflict code, I made it unused in one of my refactorings but forgot to remove